recipes/gperf/all/conanfile.py

```diff
@@ -56,6 +58,10 @@ def _configure_autotools(self):
 
     def _build_configure(self):
         with tools.chdir(self._source_subfolder):
+            try:
+                self.run("autogen.sh", run_environment=True)
```

Comment:
This script does a `wget` to update gnulib. This results in non-reproducible builds.
To fix this,
add `gnulib` to your build requirements and set the environment variable `GNULBI_TOOL` to the path of the gnulib script.

Reply:
Please also add `autoconf/2.71` as a build requirement + remove the `try/except`.
If this call fails, then the build will fail later.

Reply:
@madebr Do you mean something like this?
```
with tools.environment_append({"GNULBI_TOOL": self.deps_cpp_info["gnulib"].rootpath}):
    self.run("autogen.sh", run_environment=True)
```
> If this call fails, then the build will fail later.
If the version is not a virtual release (if it is an actual release) then there will already be a `configure` file and no `autogen.sh` file present. So my logic was, try to run it (succeeds if virtual release version) and if fails continue (succeeds if non-virtual release version). And if there is another error mode I am just not handling it how. (same as how it was before this PR)

Reply:
If you add the build requirement you can call https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/libalsa/all/conanfile.py#L44

which AFAIK is exactly what the "autogen.sh" is doing but it's more stable for Conan